dissect/target/plugins/os/unix/linux/network.py

```diff
@@ -337,6 +345,77 @@ def _parse_gateway(self, value: str | None) -> NetAddress | None:
         return ip_address(value)
 
 
+class DhclientLeaseParser(LinuxNetworkConfigParser):
+    """Parse network interfaces from dhclient DHCP ``.leases`` files.
+
+    Resources:
+        - https://linux.die.net/man/5/dhclient.conf
+    """
+
+    def interfaces(self) -> Iterator[UnixInterfaceRecord]:
+        """Parse network interfaces from dhclient DHCP ``.leases`` files."""
+
+        # TODO: Add support for renew, rebind, expire events.
+        for file in self._config_files(["/var/lib/dhclient/", "/var/lib/dhcp/"], "*.leases"):
+            try:
+                leases = configutil.parse(file, hint="leases")
+                for label, data in leases.parsed_data.items():
+                    if "lease" in label:
+                        if (fixed_address := data.get("fixed-address")) is not None:
+                            fixed_address = ip_interface(fixed_address)
+
+                        if fixed_address and (subnet_mask := data.get("option", {}).get("subnet-mask")) is not None:
+                            fixed_address = ip_interface((fixed_address.ip, subnet_mask))
```

Comment:
Slighty prefer this:

```suggestion
                        if (fixed_address := data.get("fixed-address")) is not None:
                            if (subnet_mask := data.get("option", {}).get("subnet-mask")) is not None:
                                fixed_address = ip_interface((fixed_address, subnet_mask))
                            else:
                                fixed_address = ip_interface(fixed_address)
```